src/librustc/mir/interpret/value.rs

```diff
@@ -3,16 +3,16 @@
 use ty::layout::{Align, HasDataLayout, Size};
 use ty;
 
-use super::{EvalResult, MemoryPointer, PointerArithmetic, Allocation};
+use super::{EvalResult, Pointer, PointerArithmetic, Allocation};
 
-/// Represents a constant value in Rust. ByVal and ByValPair are optimizations which
+/// Represents a constant value in Rust. ByVal and ScalarPair are optimizations which
 /// matches Value's optimizations for easy conversions between these two types
 #[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, RustcEncodable, RustcDecodable, Hash)]
 pub enum ConstValue<'tcx> {
-    /// Used only for types with layout::abi::Scalar ABI and ZSTs which use PrimVal::Undef
-    ByVal(PrimVal),
+    /// Used only for types with layout::abi::Scalar ABI and ZSTs which use Scalar::undef()
+    Scalar(Scalar),
     /// Used only for types with layout::abi::ScalarPair
-    ByValPair(PrimVal, PrimVal),
+    ScalarPair(Scalar, Scalar),
     /// Used only for the remaining cases. An allocation + offset into the allocation
     ByRef(&'tcx Allocation, Size),
```

Comment:
"Only for the remaining cases" is not true, is it? Even scalars *can be* by-ref, e.g. when a local variable has its address taken.

Though I'm also really surprised to see this is no longer a `Pointer` or so, so I guess what I know is just too outdated by now.

Reply:
Oh, never mind. This is not the `Value` type used by miri, this is something else.